core/src/main/scala/io/qbeast/core/transform/LinearTransformation.scala

```diff
@@ -137,8 +119,10 @@ case class LinearTransformation(
       case LinearTransformation(newMin, newMax, _, otherOrdering)
           if orderedDataType == otherOrdering =>
         gt(minNumber, newMin) || lt(maxNumber, newMax)
-      case IdentityToZeroTransformation(newVal) =>
+      case IdentityTransformation(newVal, otherType)
+          if otherType == orderedDataType && newVal != null =>
         gt(minNumber, newVal) || lt(maxNumber, newVal)
```

Comment:
Because it will trigger the creation of a new Revision when the identity value is not out of range.
For example, Linear(0, 10) will be superseded by Ideantity(5).

Reply:
oh, I somehow confused the terms for superseded. Is just the condition to be merged, and would only be merged if the Identity is out of the Linear range. Understood.